app/models/notification.rb

```diff
@@ -0,0 +1,31 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  belongs_to :user
+  belongs_to :course
+
+  scope :center_popup, -> { where(type: 'CenterPopup') }
+  scope :right_side_popup, -> { where(type: 'RightSidePopup') }
+  scope :email, -> { where(type: 'EmailNotification') }
+
+  def self.send_notification(user, course, options = {})
```

Comment:
hmm... I've already used "notify" as a subclass function name. If still use that may be confused...
Any other suggestion?...